src/cli/serve/serve.js

```diff
@@ -65,9 +64,10 @@ function applyConfigOverrides(rawConfig, opts, extraCliOptions) {
     delete rawConfig.xpack;
   }
 
-  if (opts.dev) {
-    set('env', 'development');
+  // only used to set cliArgs.envName, we don't want to inject that into the config
+  delete extraCliOptions.env;
```

Comment:
Some tests are passing the `--env.name=development`variable. The `env` config prefix was only present in legacy. In KP, this variable is only used in the `serve` script to initialize the `envName` cliArg, so we need to remove before merging the unknown options in the config.